ledger/src/blockstore_processor.rs

```diff
@@ -150,12 +155,14 @@ pub fn execute_batch(
     timings: &mut ExecuteTimings,
     log_messages_bytes_limit: Option<usize>,
     prioritization_fee_cache: &PrioritizationFeeCache,
+    pre_commit_callback: Option<impl FnOnce() -> Option<Option<usize>>>,
 ) -> Result<()> {
     let TransactionBatchWithIndexes {
         batch,
         transaction_indexes,
     } = batch;
     let record_token_balances = transaction_status_sender.is_some();
+    let mut transaction_indexes = transaction_indexes.to_vec();
```

Comment:
nit: adding a clone here here

Reply:
Wondering if we can remove this clone and simplify the callback interface.

What if the callback itself did an allocation (only if necessary) returning the transaction index in that case.

I also really don't like the `Option<Option<usize>>` that is there now because it hides the meaning. It's not clear from just this code that the outer option means that recording/pre-commit failed.

If I were to do this, I'd take one of these two approaches:
1. Make the outer option of return value a Result<...,()>
2. Simple enum type to more clearly represent meanings

Lean towards 1 since it's simpler, and not sure an additional enum would benefit too much here.

Reply:
> nit: adding a clone here here
>
> Wondering if we can remove this clone and simplify the callback interface.
> 
> What if the callback itself did an allocation (only if necessary) returning the transaction index in that case.

nice catch.. done: 08536f0e584c8efe6b4bb855d2319fab7dcf5ffc
I think Cow is enough. I'd like to remain the closure agnostic from allocation at all for separation of concern.
 

> I also really don't like the `Option<Option<usize>>` that is there now because it hides the meaning. It's not clear from just this code that the outer option means that recording/pre-commit failed.
> 
> If I were to do this, I'd take one of these two approaches:
> 
>     1. Make the outer option of return value a Result<...,()>
> 
>     2. Simple enum type to more clearly represent meanings
> 
> 
> Lean towards 1 since it's simpler, and not sure an additional enum would benefit too much here.

this is done: 3b852f6a948ba6c7ae36ecdab85696e8dc284023

Reply:
`Cow::to_mut` will clone if not already owned: https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.to_mut

If we want to only clone in case of `transaction_status_sender.is_some()`, it's probably the easiest/most direct to just let `TransactionBatchWithIndexes` be a mutable reference and just modify it? Clone for the actual status send.